src/Cassandra/SimpleStatement.cs

```
@@ -160,14 +160,10 @@ public SimpleStatement(IDictionary<string, object> valuesDictionary, string quer
             {
                 throw new ArgumentNullException("valuesDictionary");
             }
-            if (query == null)
-            {
-                throw new ArgumentNullException("query");
-            }
             //The order of the keys and values is unspecified, but is guaranteed to be both in the same order.
             SetParameterNames(valuesDictionary.Keys);
             base.SetValues(valuesDictionary.Values.ToArray());
-            _query = query;
+            _query = query ?? throw new ArgumentNullException("query");
```

move this to before `SetParameterNames` to fail faster